### PR TITLE
feat(138): reveal pk

### LINKED ISSUE #138: Extension - Incorporate Implicit VP when creating new keys
See https://github.com/anoma/namada/pull/592

- Extension should submit a transaction to reveal public key for the signing account on chain if not yet revealed (this could occur on account creation within the extension)
- Confirm that the signature is valid for the inner-Tx signer

NOTE: More details may be added once this is investigated further.

### PR DESCRIPTION
**Pull request by mateuszjasiuk**
Resolves #138

### CODE REVIEW

File: packages/rpc/src/RpcClient.ts [hunk 1]
```diff
@@ -25,6 +25,54 @@ class RpcClient extends RpcClientBase {
...
+    try {
+      const json: JsonRpcSuccessResponse = await this._client.execute(request);
+      const response: AbciResponse = json.result.response;
+      return response;
+    } catch (e) {
+      return Promise.reject(e);
+    }
+  }
+
+  public async get_applied_tx(hash: string): Promise<AbciResponse> {
```
Comment (jurevans) [inline packages/rpc/src/RpcClient.ts:40]: This is very nitpicky, but I would move this to camel-case - `getAppliedTx()`

File: packages/shared/lib/src/types/transaction.rs [hunk 2]
```diff
@@ -76,50 +76,33 @@ impl<'a> Transaction<'a> {
...
+        let token = Address::from_str(&token).map_err(|err| err.to_string())?;
 
         // Create and sign inner Tx
-        let tx = Tx::to_proto(
-            tx_code,
-            tx_data,
-        ).sign(&secret_key);
+        let mut tx = Tx::to_proto(tx_code, tx_data);
+
+        if sign_inner {
```
[the remark this answers is not in the document]
Comment (jurevans) [reply packages/shared/lib/src/types/transaction.rs:90]: yeah for now, I think this is perfectly fine!

**Review by jurevans — APPROVED** [review]
Just one minor comment, but this looks good!

File: packages/rpc/src/RpcClient.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (memasdeligeorgakis) [reply packages/rpc/src/RpcClient.ts]: That is fine for now!

**Review by memasdeligeorgakis — APPROVED** [review]

**Pull request merged**